core/includes/locale.inc

```diff
@@ -422,21 +422,27 @@ function locale_language_url_rewrite_url(&$path, &$options) {
   if (!isset($languages)) {
     // Get the enabled languages only.
     $languages = language_list(TRUE);
-    $languages = array_flip(array_keys($languages));
+    $languages = array_keys($languages);
   }
 
-  // Language can be passed as an option, or we go for current URL language.
+  // Array key doesn't even exist in the options.
   if (!isset($options['language'])) {
+    $options['language'] = FALSE;
```

Comment:
I don't think this is safe... it could have effects elsewhere because `$options` is passed by reference, but setting `$options['language'] = FALSE` isn't something that is expected by the `url()` function. It's documentation says that it's either `NULL` or an object.

> 'language': An optional language object. If the path being linked to is internal to the site, $options['language'] is used to look up the alias for the URL. If $options['language'] is omitted, the global $language_url will be used.

Reply:
While working on this, it turned out that `$options['language']` can be a lot of things or it doesn't even exist. (See my [comment](https://github.com/backdrop/backdrop-issues/issues/4021#issuecomment-529926814).
This really just tries to prevent "undefined index" warnings.

Anyway - re-reading the code, I see that setting it to NULL would be an option, too.
And maybe I should do the same for disabled languages (line 440).

In line 445 I check for an _object_ and ignore anything else.